test/minikube/minikube_domain_resync_test.go

```diff
@@ -465,4 +488,61 @@ func TestLoadBalancerConfigurationResync(t *testing.T) {
 	options.SetValues["admin.lbConfig.default"] = "random(first(label(node node1) any))"
 	options.SetValues["admin.lbConfig.policies.manual"] = "random(any)"
 	verifyLoadBalancer(t, namespaceName, admin0, options.SetValues)
-}
\ No newline at end of file
+}
+
+func TestKaaLimitedPermissions(t *testing.T) {
+	if os.Getenv("NUODB_LICENSE") != "ENTERPRISE" {
+		t.Skip("This test requires NuoDB 4.1.1+")
+	}
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	modifyKubeInspectorRoleInPlace(t, func(role *rbacv1.Role) {
+		// Remove "daemonsets" from the resources list
+		role.Rules[1].Resources = []string{"statefulsets", "deployments"}
```

Comment:
Also removing `pods`?

Reply:
The rule with index 1 is having normally 3 resources. This is removing just the `daemonsets` from the list.
 ```
- apiGroups:
  - apps
  resources:
  - deployments
  - statefulsets
  - daemonsets
  verbs:
  - get
  - list
  - watch
```